src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/show_app_menu_request.cc

```diff
@@ -0,0 +1,139 @@
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "sdl_rpc_plugin/commands/mobile/show_app_menu_request.h"
+
+#include "application_manager/message_helper.h"
+#include "utils/helpers.h"
+
+namespace sdl_rpc_plugin {
+using namespace application_manager;
+
+namespace commands {
+
+ShowAppMenuRequest::ShowAppMenuRequest(
+    const application_manager::commands::MessageSharedPtr& message,
+    ApplicationManager& application_manager,
+    rpc_service::RPCService& rpc_service,
+    HMICapabilities& hmi_capabilities,
+    policy::PolicyHandlerInterface& policy_handler)
+    : CommandRequestImpl(message,
+                         application_manager,
+                         rpc_service,
+                         hmi_capabilities,
+                         policy_handler) {}
+
+ShowAppMenuRequest::~ShowAppMenuRequest() {}
+
+void ShowAppMenuRequest::Run() {
+  LOG4CXX_AUTO_TRACE(logger_);
+
+  ApplicationSharedPtr app = application_manager_.application(connection_key());
+
+  if (!app) {
+    LOG4CXX_ERROR(logger_, "Application is not registered");
+    SendResponse(false, mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+    return;
+  }
+
+  if (mobile_apis::HMILevel::HMI_FULL != app->hmi_level() ||
+      helpers::Compare<mobile_apis::SystemContext::eType,
+                       helpers::NEQ,
+                       helpers::ALL>(
+          app->system_context(),
+          mobile_apis::SystemContext::SYSCTXT_MAIN,
+          mobile_apis::SystemContext::SYSCTXT_MENU)) {
+    LOG4CXX_ERROR(logger_, "Application is not activated");
+    SendResponse(false, mobile_apis::Result::REJECTED);
+    return;
+  }
+
+  smart_objects::SmartObject received_msg_params =
+      (*message_)[strings::msg_params];
+
+  smart_objects::SmartObject msg_params =
+      smart_objects::SmartObject(smart_objects::SmartType_Map);
+
+  msg_params[strings::app_id] = app->app_id();
+
+  if (received_msg_params.keyExists(strings::menu_id)) {
+    const int32_t menu_id = received_msg_params[strings::menu_id].asInt();
+    if (!app->FindSubMenu(menu_id)) {
+      LOG4CXX_ERROR(logger_, "Menu with id " << menu_id << " is not found.");
+      SendResponse(false, mobile_apis::Result::INVALID_ID);
+      return;
+    }
+    msg_params[strings::menu_id] = menu_id;
+  }
+
+  StartAwaitForInterface(HmiInterfaces::HMI_INTERFACE_UI);
+  SendHMIRequest(hmi_apis::FunctionID::UI_ShowAppMenu, &msg_params, true);
+}
+
+void ShowAppMenuRequest::on_event(const event_engine::Event& event) {
+  LOG4CXX_AUTO_TRACE(logger_);
+  const smart_objects::SmartObject& message = event.smart_object();
+
+  switch (event.id()) {
+    case hmi_apis::FunctionID::UI_ShowAppMenu: {
+      EndAwaitForInterface(HmiInterfaces::HMI_INTERFACE_UI);
+      hmi_apis::Common_Result::eType result_code =
```

Comment:
@mkorniichuk i think it's a good place for `const auto`

Reply:
@mked-luxoft fixed within ad229382f8df084c7a63027248807c7f7f03448d